lms/djangoapps/courseware/access.py

```diff
@@ -160,11 +181,18 @@ def _can_access_descriptor_with_start_date(user, descriptor, course_key):  # pyl
             descriptor,
             course_key=course_key
         )
-        return (
-            descriptor.start is None
-            or now > effective_start
-            or in_preview_mode()
-        )
+        if descriptor.start is None or now > effective_start or in_preview_mode():
+            return ACCESS_GRANTED
+
+        start_error_message = None
+        if hasattr(descriptor, 'advertised_start'):
+            if descriptor.advertised_start is not None:
```

Comment:
I put the logic for the start date message here, so `start_error_message`is the correct form of the date to show the user if the course hasn't started (ex. "July 15, 2015" or "Spring 2016"). Thus I needed `advertised_start`.